scipy/stats/_axis_nan_policy.py

```diff
@@ -31,30 +31,19 @@ def _broadcast_array_shapes(arrays, axis=None):
     return _broadcast_shapes(shapes, axis)
 
 
-def _broadcast_shapes(shapes, axis=None):
+def _process_axis(axis, shapes):
```

Comment:
This diff turned out really confusing : / Can it be moved so GitHub doesn't get so confused?

Reply:
> This diff turned out really confusing : /

Yeah, there was a lot going on behind the scenes than I thought I would need to know to make this work. Sorry! I will go over this again and simplify it as much as possible.



> I see you factored this out so it could be used below. Otherwise, is added/changed that is different from what was going on in `broadcast_shapes` before?

No, I just factored out the axis validation and sorting part. Nothing changed.

Reply:
No, it's not a problem with the code. It's just that GitHub got confused. I wouldn't refactor this expect consider moving the function so that GitHub doesn't mix up the two functions. We can move it back here later.

Reply:
Reverted for now. Will add this change back before merging.

Reply:
OK. I was going to suggest that you could just move `_process_axes` somewhere else - after `_broadcast_shapes`, for instance - so that GitHub didn't think you were heavily modifying `_broadcast_shapes`. We want it to see `_process_axes` as a totally new function, and we want it to see that some of `_broadcast_shapes` has been deleted and replaced with a call to `_process_axes`. We can manually check that `_process_axes` basically just contains all the lines deleted from `_broadcast_shapes`. Does that make sense?

Reply:
Oh, OK. Moved the _procees_axes function below _broadcast_shapes. The diff is better now.